packages/toolkit/src/query/defaultSerializeQueryArgs.ts

```diff
@@ -2,21 +2,38 @@ import type { QueryCacheKey } from './core/apiState'
 import type { EndpointDefinition } from './endpointDefinitions'
 import { isPlainObject } from '@reduxjs/toolkit'
 
+const cache: WeakMap<any, string> | undefined = WeakMap
+  ? new WeakMap()
+  : undefined
+
 export const defaultSerializeQueryArgs: SerializeQueryArgs<any> = ({
   endpointName,
   queryArgs,
 }) => {
+  let serialized = ''
+
+  const cached = cache?.get(queryArgs)
+
+  if (typeof cached === 'string') {
+    serialized = cached
+  } else {
+    const stringified = JSON.stringify(queryArgs, (key, value) =>
+      isPlainObject(value)
+        ? Object.keys(value)
+            .sort()
+            .reduce<any>((acc, key) => {
+              acc[key] = (value as any)[key]
+              return acc
+            }, {})
+        : value
+    )
+    if (isPlainObject(queryArgs)) {
+      cache?.set(queryArgs, stringified)
```

Comment:
Makes sense. There is a slight worry that two objects with the same shape will get two different entries in the map since the map is based off memory address and not value, but this shouldn't blow up in memory since I would assume old entries can get garbage collected